### ISSUE TITLE
SyntaxError: invalid syntax while using feedparser in python on Raspberry Pi

### ISSUE THREAD

**Issue by teacow2** [kurtmckee/feedparser#256]
I have installed feedparser on my Raspberry Pi, but when I run my python program using "import feedparser", I get this error:  

Traceback (most recent call last):
  File "rrsTest.py", line 1, in <module>
    import feedparser
  File "/usr/local/lib/python3.5/dist-packages/feedparser/__init__.py", line 31, in <module>
    from .api import parse
  File "/usr/local/lib/python3.5/dist-packages/feedparser/api.py", line 56, in <module>
    from . import http
  File "/usr/local/lib/python3.5/dist-packages/feedparser/http.py", line 156
    new_pieces[1] = f'{url_pieces.hostname}:{url_pieces.port}'
                                                             ^
SyntaxError: invalid syntax

I have successfully used the same program with feedparser in python in Windows, so I don't understand why the module isn't working here.

**Comment by kurtmckee** [comment]
Hi Nicholas! The latest version of feedparser doesn't support Python 3.5. However, I botched the 6.0.0 and 6.0.1 releases in a way that allowed then to be installed on unsupported Python versions. What version of feedparser is currently installed? You can check by importing feedparser and then accessing `feedparser.__version__`.

You might double-check if there's a newer version of Python available for your RPi. Which OS and version are you running?

**Comment by teacow2** [comment]
Ah, ok. I have feedparser 6.0.0 installed on the pi. I was running Python
3.5 on a Debian Linux 9.



Sorry to waste your time here. I was under the mistaken impression that
Python updated in a general pi apt-get update, but I’ve now updated to 3.9,
and it seems to be working fine.



Nicholas Fortugno

CCO/Co-Founder, Playmatics LLC

nick@playmatics.com

(347) 731-0894

www.playmatics.com

Pronouns: he/him/his



*From:* Kurt McKee <notifications@github.com>
*Sent:* Wednesday, December 30, 2020 8:12 PM
*To:* kurtmckee/feedparser <feedparser@noreply.github.com>
*Cc:* Nicholas Fortugno <nick@playmatics.com>; Author <
author@noreply.github.com>
*Subject:* Re: [kurtmckee/feedparser] SyntaxError: invalid syntax while
using feedparser in python on Raspberry Pi (#256)



Hi Nicholas! The latest version of feedparser doesn't support Python 3.5.
However, I botched the 6.0.0 and 6.0.1 releases in a way that allowed then
to be installed on unsupported Python versions. What version of feedparser
is currently installed? You can check by importing feedparser and then
accessing `feedparser.__version__`.

You might double-check if there's a newer version of Python available for
your RPi. Which OS and version are you running?

On December 31, 2020 12:28:07 AM UTC, Nicholas Fortugno <
notifications@github.com> wrote:
>I have installed feedparser on my Raspberry Pi, but when I run my python
program using "import feedparser", I get this error:
>
>Traceback (most recent call last):
> File "rrsTest.py", line 1, in <module>
> import feedparser
> File "/usr/local/lib/python3.5/dist-packages/feedparser/__init__.py",
line 31, in <module>
> from .api import parse
> File "/usr/local/lib/python3.5/dist-packages/feedparser/api.py", line 56,
in <module>
> from . import http
> File "/usr/local/lib/python3.5/dist-packages/feedparser/http.py", line 156
> new_pieces[1] = f'{url_pieces.hostname}:{url_pieces.port}'
> ^
>SyntaxError: invalid syntax
>
>I have successfully used the same program with feedparser in python in
Windows, so I don't understand why the module isn't working here.
>
>
>--
>You are receiving this because you are subscribed to this thread.
>Reply to this email directly or view it on GitHub:
>https://github.com/kurtmckee/feedparser/issues/256

—
You are receiving this because you authored the thread.
Reply to this email directly, view it on GitHub
<https://github.com/kurtmckee/feedparser/issues/256#issuecomment-752808117>,
or unsubscribe

**Comment by kurtmckee** [comment]
Nicholas, no time was wasted. It looks like you picked up feedparser 6.0.0 during the window between when it was released and when I yanked it from PyPI. I'm sorry for the trouble, but I'm glad that it's working now.

Happy 2021!

**Comment by teacow2** [comment]
No worries at all. And thanks for replying so quickly. This is definitely
the best support I’ve ever gotten from GitHub in my life. Happy New Year!



Nicholas Fortugno

CCO/Co-Founder, Playmatics LLC

nick@playmatics.com

(347) 731-0894

www.playmatics.com

Pronouns: he/him/his



*From:* Kurt McKee <notifications@github.com>
*Sent:* Friday, January 1, 2021 3:33 PM
*To:* kurtmckee/feedparser <feedparser@noreply.github.com>
*Cc:* Nicholas Fortugno <nick@playmatics.com>; Author <
author@noreply.github.com>
*Subject:* Re: [kurtmckee/feedparser] SyntaxError: invalid syntax while
using feedparser in python on Raspberry Pi (#256)



Nicholas, no time was wasted. It looks like you picked up feedparser 6.0.0
during the window between when it was released and when I yanked it from
PyPI. I'm sorry for the trouble, but I'm glad that it's working now.

Happy 2021!

—
You are receiving this because you authored the thread.
Reply to this email directly, view it on GitHub
<https://github.com/kurtmckee/feedparser/issues/256#issuecomment-753378949>,
or unsubscribe
